backend/curation/api/v1/curation/datasets/actions.py

```diff
@@ -0,0 +1,33 @@
+from flask import jsonify, make_response
+
+from backend.curation.api.v1.curation.collections.common import reshape_dataset_for_curation_api
+from backend.layers.common.entities import CollectionLinkType
+from backend.portal.api.providers import get_business_logic
+
+
+def get():
+    """
+    Datasets index endpoint to retrieve full metadata for all public Datasets.
+    """
+    collections_with_datasets = get_business_logic().get_all_published_collections_with_datasets()
+
+    all_datasets_with_collection_name_and_doi = []
+
+    for collection in collections_with_datasets:
+        assert collection.datasets  # Any public Collection should have at least one Dataset
```

Comment:
I think this assert belongs in a test rather than in the API code. We already have checks at publish time that wouldn't let you publish collections with 0 datasets.

Reply:
I apologize. I pushed this change to rdev and forgot to push it to the feature branch 🤦‍♂️